### PR TITLE
Rootless fixes

### PR DESCRIPTION
**Pull request by captinc**
### All Submissions

* [ ] Have you made sure there aren't any other open [Pull Requests](https://github.com/ProcursusTeam/Procursus/pulls) for the same update/change?
* [ ] This Pull Request doesn't contain any package additions; it's a small change (e.g README change)

### Package Additions/Updates

* [ ] Have you confirmed this builds & works as intended on an iOS device (if applicable)?
* [ ] Have you confirmed this builds & works as intended on a macOS device (if applicable)?

### Details

* Various fixes for build errors
* Various fixes for rootless path issues
* Various updates to package versions

Note: compiling fails when using Xcode Command Line Tools 13.3 and newer - you have to use 13.2.1 or older

### CODE REVIEW

**Comment by captinc** [comment]
Sidenote, compilation fails with Xcode Command Line Tools 13.3 and newer. You have to use 13.2.1 or older

File: makefiles/lsof.mk [hunk 1]
```diff
@@ -5,12 +5,14 @@ endif
...
+LSOF_VERSION := 70
 DEB_LSOF_V   ?= $(LSOF_VERSION)
 
 lsof-setup: setup
-	wget -q -nc -P $(BUILD_SOURCE) https://opensource.apple.com/tarballs/lsof/lsof-$(LSOF_VERSION).tar.gz
-	$(call EXTRACT_TAR,lsof-$(LSOF_VERSION).tar.gz,lsof-$(LSOF_VERSION),lsof)
+	$(call GITHUB_ARCHIVE,apple-oss-distributions,lsof,$(LSOF_VERSION),lsof-$(LSOF_VERSION))
+	$(call EXTRACT_TAR,lsof-$(LSOF_VERSION).tar.gz,lsof-lsof-$(LSOF_VERSION),lsof)
+	$(call DO_PATCH,lsof,lsof,-p1)
+
```
Comment (TheRealKeto) [inline makefiles/lsof.mk:15]: Delete this blank line

```suggestion
	$(call DO_PATCH,lsof,lsof,-p1)
```

File: makefiles/network-cmds.mk [hunk 2]
```diff
@@ -97,19 +97,24 @@ network-cmds:
...
+		$(CC) $(LDFLAGS) -arch $(MEMO_ARCH) -isysroot $(TARGET_SYSROOT) $(PLATFORM_VERSION_MIN) -isystem include -o $$tproj $$tproj.tproj/!(ns).c ecnprobe/gmt2local.c -DPRIVATE -DINET6 -DPLATFORM_iPhoneOS -D__APPLE_USE_RFC_3542=1 -DUSE_RFC2292BIS=1 -D__APPLE_API_OBSOLETE=1 -DTARGET_OS_EMBEDDED=1 -Dether_ntohost=_old_ether_ntohost; \
 	done
 	cp -a $(BUILD_WORK)/network-cmds/kdumpd $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/libexec
 	cp -a $(BUILD_WORK)/network-cmds/{arp,ndp,traceroute,mnc,mtest,traceroute6,ifconfig,ip6addrctl,netstat,ping6,route,rtsol} $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin
 	cd $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin; \
 	for bin in ifconfig ip6addrctl netstat ping6 route rtsol; do \
-		$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/$$bin $(BUILD_STAGE)/network-cmds/sbin; \
+		$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/$$bin $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)sbin; \
```
Comment (TheRealKeto) [inline makefiles/network-cmds.mk:115]: Use the correct prefix, per suggestion

```suggestion
		$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/$$bin $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin; \
```

File: makefiles/network-cmds.mk [hunk 3]
```diff
@@ -97,19 +97,24 @@ network-cmds:
...
 	done
-	$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/ping6 $(BUILD_STAGE)/network-cmds/bin
+	$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/ping6 $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)bin
```
Comment (TheRealKeto) [inline makefiles/network-cmds.mk:117]: Again, use the correct prefix...

```suggestion
	$(LN_S) ../$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/sbin/ping6 $(BUILD_STAGE)/network-cmds/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
```

**Review by TheRealKeto — CHANGES_REQUESTED** [review]
Just some QOL suggestions; did you test other targets as well? I fear some of these changes will have negative effects on other targets.

**Comment by captinc** [comment]
Good point, I have not tested other targets; QOL suggestions added

**Pull request closed without merging**